ament_cmake_pclint/cmake/ament_pclint.cmake

```diff
@@ -0,0 +1,145 @@
+# Copyright 2017 Apex.AI, Inc.
+#
+#
+# This file contains modified code from the following open source projects
+# published under the licenses listed below:
+#
+#
+# # Copyright 2014-2015 Open Source Robotics Foundation, Inc.
+# #
+# # Licensed under the Apache License, Version 2.0 (the "License");
+# # you may not use this file except in compliance with the License.
+# # You may obtain a copy of the License at
+# #
+# #     http://www.apache.org/licenses/LICENSE-2.0
+# #
+# # Unless required by applicable law or agreed to in writing, software
+# # distributed under the License is distributed on an "AS IS" BASIS,
+# # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# # See the License for the specific language governing permissions and
+# # limitations under the License.
+
+function(_parse_pclint_input)
+
+  find_program(ament_pclint_BIN NAMES "ament_pclint")
+  if(NOT ament_pclint_BIN)
+    message(FATAL_ERROR "ament_pclint() could not find program 'ament_pclint'")
+  endif()
+
+  set(cmd "${ament_pclint_BIN}" ${ARG_UNPARSED_ARGUMENTS})
+
+  get_property(INCLUDED_DIRS DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+
+  if(INCLUDED_DIRS)
+      list(APPEND cmd "--include-directories" ${INCLUDED_DIRS})
+      set(_INCLUDE_DIRS_FLAG_SET "Found")
+  endif()
+
+  # Include 'include' folders in AMENT_PREFIX_PATH
+  if(DEFINED ENV{AMENT_PREFIX_PATH})
+    if(NOT _INCLUDE_DIRS_FLAG_SET)
+      list(APPEND cmd "--include-directories")
+      set(_INCLUDED_DIRS_FLAG_SET "Found")
+    endif()
+    set(AMENT_PATHS_STR $ENV{AMENT_PREFIX_PATH})
+    # In Linux, path lists are separated by colons ":",
+    # while in Windows (like in cmake) they are separated by semi-colons ";"
+    if(NOT WIN32)
+        string(REPLACE ":" ";" AMENT_PATHS_LIST ${AMENT_PATHS_STR})
+    else()
+        set(AMENT_PATHS_LIST ${AMENT_PATHS_STR})
+    endif()
+    foreach(path ${AMENT_PATHS_LIST})
+        list(APPEND cmd "${path}/include")
+    endforeach()
+  endif()
+
+  get_directory_property(COMPILE_DEFS DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} COMPILE_DEFINITIONS)
+
+  if(COMPILE_DEFS)
+    list(APPEND cmd "--compiler-definitions" "${COMPILE_DEFS}")
+  endif()
+
+  if(ARG_LANGUAGE)
+    list(APPEND cmd "--language" "${ARG_LANGUAGE}")
+  endif()
+
+  if(ARG_LNT_FILE)
+    get_filename_component(config_dir ${ARG_LNT_FILE} DIRECTORY)
+    list(APPEND cmd "--pclint-config-dir" ${config_dir})
+    get_filename_component(config_file ${ARG_LNT_FILE} NAME)
+    list(APPEND cmd "--pclint-config-file" ${config_file})
+  endif()
+
+  set(cmd ${cmd} PARENT_SCOPE)
+
```

Comment:
Nitpick: remove the empty lines right after `function()` and before `endfunction()`.

Reply:
Done